addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/CsrfTokenScanRule.java

```diff
@@ -186,8 +187,13 @@ public void scan() {
                     }
                 }
                 newMsg.setCookieParams(newCookies);
-                sendAndReceive(newMsg);
+                sendAndReceive(newMsg, false);
 
+                if (newMsg.getResponseHeader().getStatusCode() != 200) {
+                    LOGGER.debug("Skipping form {} due to non-200 response code", formIdx);
+                    formIdx++;
+                    continue;
+                }
```

Comment:
I think this might be a bit too presumptive. Anything in the 200-299 range is success. Plus there are plenty of models and frameworks that use redirects as a result of form submissions.